crates/rust/src/lib.rs

```diff
@@ -1105,6 +1106,17 @@ impl WorldGenerator for RustWasm {
         id: InterfaceId,
         _files: &mut Files,
     ) -> Result<()> {
+        let mut to_define = Vec::new();
+        for (name, ty_id) in resolve.interfaces[id].types.iter() {
+            let full_name = full_wit_type_name(resolve, *ty_id);
+            if let Some(TypeGeneration::Remap(_)) = self.with.get(&full_name) {
```

Comment:
Could we add an `impl TypeGeneration { fn generated(&self) -> bool {...} }` and use that here?

Reply:
Sure. Done in https://github.com/bytecodealliance/wit-bindgen/pull/1173/commits/613d2fd95f360e554613d435e74c1c52b85617bf